src/test/py/peer/peer_test.py

```diff
@@ -202,72 +202,238 @@ def test_verify_integrity(self):
         self.ledger.queue[0] = (self.peer, b"corrupted_hash")
         self.assertFalse(self.ledger.verify_integrity())
 
-    async def test_announce_trade(self):
+    async def test_handle_exchange_announcement_deadline_trade(self):
+        """
+        Test case for the handle_exchange_announcement_deadline method for the trade exchange_type
+        of the Peer class.
+        """
+
+        announcement_type = "trade"
+        announcement_key = "key"
+        offer_announcement = OfferAnnouncement("trade", 0, "artwork")
+
+        await self.peer.handle_exchange_announcement_deadline(
+            announcement_type, announcement_key, offer_announcement
+        )
+
+        self.assertNotIn(announcement_key, self.peer.inventory.pending_exchanges)
+        self.assertIn(announcement_key, self.peer.inventory.completed_exchanges)
+        self.peer.node.set.assert_called_once_with(
+            announcement_key, pickle.dumps(offer_announcement)
+        )
+        self.peer.logger.info.assert_any_call("Exchange announcement deadline reached")
+        self.peer.logger.info.assert_any_call("%s announced", announcement_type)
+
+    async def test_handle_exchange_announcement_deadline_sale(self):
+        """
+        Test case for the handle_exchange_announcement_deadline method for the sale exchange type
+        of the Peer class.
+        """
+
+        announcement_type = "trade"
+        announcement_key = "key"
+        offer_announcement = OfferAnnouncement("trade", 10, "artwork")
```

Comment:
Similar to above, shouldn't it be`sale`? Or, am I missing something?
... looking the logic over, looks like that attribute is never used during processing.